app/components/autocomplete/date_suggestion/index.js

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+
+import {getMatchTermForDateMention} from 'app/selectors/autocomplete';
+import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+
+import DateSuggestion from './date_suggestion';
+
+function mapStateToProps(state, ownProps) {
+    const {cursorPosition, value} = ownProps;
+
+    const newValue = value.substring(0, cursorPosition);
+    const matchTerm = getMatchTermForDateMention(newValue);
```

Comment:
I'm not very familiar with the mobile autocomplete code. Are there ever multiple instances of Autocomplete/DateSuggestion existing at a single time? If so, you'll have two DateSuggestions with calling `getMatchTermForDateMention` with different values of `newValue`, so they won't be memoizing anything.

If that's the case, you can replace `getMatchTermForDateMention` with a factory function that returns a new copy of `getMatchTermForDateMention` and then replace this `mapStateToProps` with a similar factory function like
```javascript
import {makeGetMatchTermForDateMention} from 'app/selectors/autocomplete';

function makeMapStateToProps() {
    const getMatchTermForDateMention = makeGetMatchTermForDateMention();
    return (state, ownProps) => {
        const newValue = ...;
        return {
            matchTerm: getMatchTermForDateMention(newValue);
        };
    };
}

export default connect(makeMapStateToProps)(DateSuggestion);
```

Reply:
no, there's only 1 DateSuggestion component at any given time

Reply:
this follows the exact same pattern as the other types of suggestions in the mobile project

Reply:
It looks like there can be multiple Autocompletes between the PostTextbox and the Search screen. If the other suggestion components do it this way, I can create a ticket to change those, but I'd rather do this one correctly.